### ISSUE TITLE
Add ceramic network configurations

### ISSUE THREAD

**Issue by stbrody** [ceramicnetwork/js-ceramic#573]
Add various network configurations to js-ceramic.  Depending on the network specified we will set the allow list of chainIds for anchor records and the pub/sub topic used by the dispatcher accordingly

**Comment by stbrody** [comment]
The [chainId discussion doc](https://www.notion.so/threebox/ChainId-discussion-f4524c2d5cc54f8b947b1ed1b2e4b8d9?d=fe786f3b-575f-45a0-8975-9950ff9f3909#3897b52d268940f4aac15d61f0597ae3) says to add a random number to the pubsub topic used when the `local` or `inmemory` network is specified.  This causes problems, however, for example in our tests which are all on the `inmemory` network - two different Ceramic instances created will be on different pubsub topics and thus won't receive each others messages.  So we then either need to expose the 'random' number to be a config option so that usable `local` and `inmemory` setups can be meaningfully created, or we get rid of the random part.

Adding the number as a config option increases the friction to getting quickly started running a local Ceramic node to test against and use to play with the API.  You'd need to pick a random number by default and log it to the console so that developers and operators can know what it is, while allowing the number to be provided explicitly both on the CLI and in the CeramicConfig object so that it's possible to make Ceramic instances on the same `local` or `inmemory` network as another.  Getting rid of the random part however means that test deployments that are accessible on the public internet might wind up connecting to other test deployments and sending each other messages, reducing the degree to which test deployments are completely isolated and local.

I'm ever so slightly leaning toward taking the random number out to make the initial developer experience smoother, and since running test deployments that are on the public internet will wind up connecting to other deployments anyway, even if they don't gossip each others messages, and we expect most test deployments will be behind a firewall and/or NAT anyway.  But my preference is slight and I could easily be swayed, so I'm interested in other people's input here. @oed @simonovic86 @PaulLeCam @michaelsena

**Comment by oed** [comment]
Maybe we can make sure that when you run clayground the IPFS instance doesn't connect to external internet? @simonovic86

**Comment by PaulLeCam** [comment]
I think the `local` mode should be having a standalone node. If developers wants to connect their local node to a given network that should be possible setting a flag with the CLI, but by default it should be isolated.

Regarding the `inmemory`/tests setup I agree it should be possible to connect multiple nodes to the same network to run tests on a cluster, so maybe we can use a default value if nothing is provided, but letting the developers provide the number or even string they want to use for the pubsub channel (always prefixed by `inmemory-`) by API call option, CLI flag and/or environment variable?

**Comment by stbrody** [comment]
Since the PR that got merged uses deterministic pub/sub channels for everything, I filed https://github.com/ceramicnetwork/js-ceramic/issues/581 to make `local` and `inmemory` isolated by default.  I think for now we should hold off on adding any user-visible way to control the pubsub topic for the daemon.  If/when we decide that it's useful for users to be able to run `local` or `inmemory` nodes that talk to each other, we can add it then.  We will, however, need an internal way to control this so our tests that use `inmemory` can stay connected to each other.  Does that sound good @oed @PaulLeCam?
